### PR TITLE
Manually add a COMPLETELY NEW book to database

### PR DESCRIPTION
**Pull request by gradiants93**
Can add a new user collection record to database. 
MUST use a NEW book -> creates a format based on book id pk -> creates the user collection record based on book format id pk.

### CODE REVIEW

File: client/src/pages/add_new_book.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    if (value && typeof value === "string") {
+      if (value.toLowerCase() === "true") return true;
+      if (value.toLowerCase() === "false") return false;
+    }
+    return value;
+  };
+
+  //create functions that handle the event of the user typing into the form
+  const handleOwnedChange = (event) => {
+    // console.log(str2bool(event.target.value));
```
Comment (jberardini) [inline client/src/pages/add_new_book.js]: remove

File: client/src/pages/add_new_book.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  const handleAuthorFChange = (event) => {
+    const author_f = event.target.value;
+    setBook((book) => ({ ...book, author_f }));
+  };
+
+  const handleAuthorLChange = (event) => {
+    const author_l = event.target.value;
+    setBook((book) => ({ ...book, author_l }));
+  };
+  const handleTitleChange = (event) => {
```
Comment (jberardini) [inline client/src/pages/add_new_book.js]: these handleChange methods are duplicative. I would suggest something like

`const handleChange = (event, fieldName, isBoolean=false) => {
    const parsedValue = isBoolean ? str2bool(event.target.value) : event.target.value;
    setBook((book) => ({...book, [fieldName]: parsedValue})
`

File: client/src/pages/add_new_book.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+  // The handle submit function now needs logic for adding to DB and using correct ids
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log(book);
+    postNewBook(book);
+  };
+  return (
+    <div
+      style={{
```
Comment (jberardini) [inline client/src/pages/add_new_book.js:55]: assuming you'll move these into a stylesheet once the design is more finalized?

File: server/server.js [hunk 4]
```diff
@@ -18,54 +19,54 @@ app.get("/", (req, res) => {
...
+      id: 5,
+      title: "",
+      author_f: "",
+      author_l: "",
+      format: "",
+      owned: "",
+      read: "",
+    },
+  ];
+  // res.json(mockBooks);
```
Comment (jberardini) [inline server/server.js]: remove?

**Review by jberardini — COMMENTED** [review]
looking good! only other thing I would suggest is to remove the console.log statements throughout, assuming you're not actively using them for debugging

File: client/src/pages/add_new_book.js (hunk 3 above)
Comment (EileenLong) [reply client/src/pages/add_new_book.js:55]: Ya, right now it was because it was showing funky on the page

File: server/server.js (hunk 4 above)
Comment (EileenLong) [reply server/server.js]: The mockData and response or just the response?

**Pull request merged**